src/components/DistanceMapView/types.ts

```diff
@@ -0,0 +1,8 @@
+import {StyleProp, ViewStyle} from 'react-native';
+import {ComponentProps} from '@components/MapView/types';
+
+type DistanceMapViewProps = ComponentProps & {
+    overlayStyle: StyleProp<ViewStyle>;
+};
+
+export default DistanceMapViewProps;
```

Comment:
Since `ComponentProps` consists of `MapViewProps & MapViewOnyxProps`, and we don't pass `MapViewOnyxProps` `userLocation` to the component (you can test usage of `DistanceMapView` component, with current implementation it will give TS error requiring `userLocation` param), I think we should use `MapViewProps` only.
```suggestion
import {StyleProp, ViewStyle} from 'react-native';
import type {MapViewProps} from '@components/MapView/MapViewTypes';

type DistanceMapViewProps = MapViewProps & {
    overlayStyle: StyleProp<ViewStyle>;
};

export default DistanceMapViewProps;
```
Besides that we need to update [MapView index.tsx](https://github.com/Expensify/App/blob/main/src/components/MapView/index.tsx#L5) to use `MapViewProps` only as well.

This way the combination of `MapViewProps & MapViewOnyxProps` will be used only in MapView component files ([1](https://github.com/Expensify/App/blob/main/src/components/MapView/MapView.tsx),[2](https://github.com/Expensify/App/blob/main/src/components/MapView/MapView.website.tsx)) which wrapped into withOnyx HOC and have the Onyx values.

Reply:
@VickyStash Made the requested change here: https://github.com/Expensify/App/pull/33553/commits/975ebad50310dfca667ef10c5e0c2c252ab6c64b
Please resolve the conversation if it is as expected 😇